scipy/sparse/csgraph/__init__.py

```diff
@@ -27,6 +27,8 @@
    reverse_cuthill_mckee -- compute permutation for reverse Cuthill-McKee ordering
    maximum_flow -- solve the maximum flow problem for a graph
    maximum_bipartite_matching -- compute a maximum matching of a bipartite graph
+   minimum_weight_full_bipartite_matching - compute a minimum weight full
+       matching of a bipartite graph
```

Comment:
This is what the doc build will be complaining about - if you put it all on a single line it should work

Reply:
Thanks for checking. This was actually my first hunch as well, but rebuilding the docs locally after making the change gave me the same warning (in addition to now violating PEP8). Let me again though.

Reply:
Yep, you're right, that turned the test green. How do you feel about the > 80 char line? (Really, `minimum_weight_full_bipartite_matching` is awkwardly long, but hard to shorten without making it imprecise; perhaps `minimum` could be `min`, but that's about it.)

Reply:
As long as the rendered html doc looks fine, then it's not a problem. 

Yes, the function name is very long - may be worth shortening. `minimum` -> `min`, maybe also dropping `full`?

Reply:
FWIW, I chose `minimum` for consistency with `maximum_bipartite_matching` and `maximum_flow`, but I have no strong opinion (you could argue that `min` is more consistent with the built-in and NumPy ...).

Regarding `full`, the mathematician in me likes it since if you wanted to be pedantic, you would argue that the empty matching is a "minimum weight bipartite matching" whenever weights are positive, and that's not what we want. Conversely, you could argue the same for "minimum cost flow"; there, you input the minimum acceptable flow, then minimize over all flows satisfying that demand, yet the demand makes no appearance in the name. I don't think there's an equally standard naming convention in this case, so the choice to go with Karp's naming is somewhat arbitrary. It does also match up with NetworkX's [`nx.algorithms.bipartite.minimum_weight_full_matching`](https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.bipartite.matching.minimum_weight_full_matching.html) (but mainly because I got to pick the name there as well). At the end of the day, I don't have a strong opinion here either, but am slightly biased towards having `full`, simply to avoid the slight risk of potentially confusing newcomers if it's not there.

Reply:
I'm in favour of dropping `full` from the function name, and leaving the explanation that the matching is full in the docstring. Most literature that I have seen on this topic just calls it the "minimum weight bipartite matching" problem.

Also, isn't "maximum" more common than "full", as in "minimum weight maximum bipartite matching"?

Reply:
Don't think I ever saw "maximum" in use; I think that would also be a bit misleading: In "maximum bipartite matching", one just looks for a matching of maximum cardinality among all matching, and there's no requirement that the matching has the cardinality of one of the partitions; it just has to be as large as possible.

Here, we explicitly require that the each vertex in the smaller of the two partitions is incident to an edge in the matching. So a full matching is necessarily maximal, but not the other way around. One common piece of terminology seems to be "minimum weight perfect matching", which would be accurate here if we only considered the square case; in the rectangular case, no perfect matchings exist, so the best we can do is ask for a full one.

Of course, that doesn't mean that asking for a "minimum weight maximum bipartite matching" is not doable; it's just a more general problem than what we solve here.

Reply:
Note that the Travis CI Lint check [is picking up](https://travis-ci.org/github/scipy/scipy/jobs/743384625#L486) the long line as well.